### PR TITLE
Test LIMIT pushdown

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: presto-memsql/src/test/java/io/prestosql/plugin/memsql/TestMemSqlIntegrationSmokeTest.java [hunk 1]
```diff
@@ -181,6 +182,19 @@ public void testInsertIntoNotNullColumn()
...
+    @Test
+    public void testLimitPushdown()
+    {
+        assertThat(query("SELECT name FROM nation LIMIT 30")).isCorrectlyPushedDown(); // Use high limit for result determinism
+
+        // with filter over numeric column
+        assertThat(query("SELECT name FROM nation WHERE regionkey = 3 LIMIT 5")).isCorrectlyPushedDown();
+
+        // with filter over varchar column
+        // TODO (https://github.com/prestosql/presto/issues/5263) should be `.isNotFullyPushedDown(FilterNode.class)`
```
Comment (findepi) [inline presto-memsql/src/test/java/io/prestosql/plugin/memsql/TestMemSqlIntegrationSmokeTest.java:194]: https://github.com/prestosql/presto/issues/5263

File: presto-druid/src/test/java/io/prestosql/plugin/druid/TestDruidIntegrationSmokeTest.java [hunk 2]
```diff
@@ -272,4 +272,17 @@ public void testFilteringForTablesAndColumns()
...
+    public void testLimitPushdown()
+    {
+        assertThat(query("SELECT name FROM nation LIMIT 30")).isCorrectlyPushedDown(); // Use high limit for result determinism
+
+        // with filter over numeric column
+        assertThat(query("SELECT name FROM nation WHERE regionkey = 3 LIMIT 5")).isCorrectlyPushedDown();
+
+        // with filter over varchar column
+        assertThat(query("SELECT name FROM nation WHERE name < 'EEE' LIMIT 5")).isCorrectlyPushedDown();
+    }
```
Comment (kokosing) [inline presto-druid/src/test/java/io/prestosql/plugin/druid/TestDruidIntegrationSmokeTest.java:286]: what about aggregation and project? We have at the moment 4 variables (predicate, limit, projection and aggregation). 
Having dedicated tests per each pushdown gives 4 tests, for pairs another 12 tests, for triples another test 4 tests, and for all cases at once another 1 tests. 21 in total.

What are your thoughts to tackle that?

**Review by kokosing — APPROVED** [review]

File: presto-druid/src/test/java/io/prestosql/plugin/druid/TestDruidIntegrationSmokeTest.java (hunk 2 above)
Comment (findepi) [reply presto-druid/src/test/java/io/prestosql/plugin/druid/TestDruidIntegrationSmokeTest.java:286]: > what about aggregation and project?

agg pushdown with limit does not work today -- https://github.com/prestosql/presto/issues/5261

> What are your thoughts to tackle that?

i want to organize pushdown test around top level thing that is pushed down (eg agg pushdown tests won't cover LIMIT (applied after agg), unless the LIMIT is in subquery)
i also want to order them by the ordering in which they are appledi in JdbcTableHandle

File: presto-druid/src/test/java/io/prestosql/plugin/druid/TestDruidIntegrationSmokeTest.java [hunk 3]
```diff
@@ -272,4 +272,17 @@ public void testFilteringForTablesAndColumns()
         actualColumns = computeActual("DESCRIBE " + datasourceB);
         Assert.assertEquals(actualColumns, expectedColumns);
     }
+
+    @Test
```
Comment (sopel39) [inline presto-druid/src/test/java/io/prestosql/plugin/druid/TestDruidIntegrationSmokeTest.java:276]: why not to add basic test case to `io.prestosql.plugin.jdbc.TestJdbcIntegrationSmokeTest`?

Comment (findepi) [reply presto-druid/src/test/java/io/prestosql/plugin/druid/TestDruidIntegrationSmokeTest.java:276]: `TestJdbcIntegrationSmokeTest` is not generic
did you mean `AbstractTestIntegrationSmokeTest`?

i thought about that. 
Some connnectors do not support limit pushdown, some support partially (eg mysql & varchars)....
Yet, i am OK if we revisit this later.
would you be ok to coninue with current approach?

**Review by sopel39 — APPROVED** [review]

**Pull request merged**